### DISCUSSION TITLE
Central Token area

### DISCUSSION THREAD

**Discussion by JaySMa** [dnnsoftware/Dnn.Platform#4221] [Q&A] [answered]
Hello,
Is there a possibility of adding the use of Tokens from a central location?

E.g., we're a county government.  There are multiple pages containing information for the commissioners and other elected officials.  Obviously, they come and go and it's sometimes hard to track down every page where their name is listed, etc.

If the system could have a central location which we could create custom tokens for the information.  E.g. [Comm1Name] = "Joe Smith"; [Comm1Picture] = '/portals/0/images/JoeSmith.png'; [Comm1Term] = 'January 1, 2016 through January 1, 2020'

Then within the HTML module, we could enter the Token name and the information could be auto-filled.  E.g.
[Comm1Picture]
[Comm1Name]
Term: [Comm1Term]

[Joe Smith Image]
Joe Smith
Term: January 1, 2016 through January 1, 2020

Then to add to this, the ability to share the information across multiple portals.  Checkboxes which would allow us to select which portal the information can be displayed on.

Thank you.

**Comment by bdukes** [comment, marked as answer]
There isn't a mechanism to do this right now.  #3820 moved token logic to a provider model in DNN 9.7.0.  It was added by DNN Sharp (@bogdan-litescu) and, I assume, added to support their [My Tokens module](https://www.dnnsharp.com/dnn/modules/my-custom-tokens), which does what you're asking (before this change they had to make adjustments to core components to make their tokens available).  So you'll probably want to look at that module.  Alternatively, you could build your own Token Provider which would provide a place for you to add your custom tokens in a way that's accessible to the HTML module (among others).
